### PR TITLE
Closes #358

### LINKED ISSUE #358: `test-Rules-validity` - add few more specific test.
A few more tests should be added for the edge case. For instance:

- [x] if a slot of a class stores a probability value without `0`, `1` bounds, there should be a tests checking if the validation function reports a proper error for these edge case.
- [x] Also, if `0` or `1` is allowed, then it should be checked that the validation function passes when user tries to specify these edge values.
- [x] Also adjust the help for classes for same cases.

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #358

### CODE REVIEW

File: R/Rules-class.R [hunk 1]
```diff
@@ -35,7 +35,8 @@ setClass(
 #'
 #' [`NextBestMTD`] is the class for next best dose based on MTD estimate.
 #'
-#' @slot target (`proportion`)\cr target toxicity probability.
+#' @slot target (`proportion`)\cr target toxicity probability, except \eqn{0} or
+#'   \eqn{1}.
```
Comment (danielinteractive) [inline R/Rules-class.R:39]: for maintenance purpose, highly recommend to just say 1 or 0 without the markup around it.

File: R/Rules-class.R [hunk 2]
```diff
@@ -709,9 +712,9 @@ setClass(
 #' @slot intervals (`numeric`)\cr a vector with the left bounds of the relevant
 #'   intervals. This parameters specifies the right bounds of the intervals.
 #'   For example, `intervals  = c(0, 50, 100)` specifies three intervals:
-#'   (0, 50), [50, 100) and [100, +Inf). That means, the right bound of the
-#'   intervals are exclusive to the interval and the last interval goes from the
-#'   last value to infinity.
+#'   \eqn{(0, 50)}, \eqn{[50, 100)} and \eqn{[100, +Inf)}. That means, the right
```
Comment (danielinteractive) [inline R/Rules-class.R:715]: here it is useful and ok

File: R/Rules-class.R [hunk 3]
```diff
@@ -761,21 +764,24 @@ IncrementsRelative <- function(intervals, increments) {
...
+#' dose in case of a DLT in part \eqn{1}, or `clean_start` shift (e.g. \eqn{0})
+#' in case of no DLTs in part \eqn{1}.
+#' If part \eqn{1} will still be on in the
+#' next cohort, then the next dose level will be the next higher dose level in
+#' the `part1Ladder` slot of the data object.
+#' If part \eqn{2} has been started before, the usual relative increment rules
+#' apply, see [`IncrementsRelative`].
+#'
+#' @slot dlt_start (`count`)\cr the dose level increment for starting part
+#'   \eqn{2} in case of a DLT in part \eqn{1}.
```
Comment (danielinteractive) [inline R/Rules-class.R:779]: also here, just 2 and 1

**Review by danielinteractive — COMMENTED** [review]
thanks Wojciech, please see inside

File: R/Rules-class.R (hunk 1 above)
Comment (wwojciech) [reply R/Rules-class.R:39]: ohhh, sure, then It would be a lot o changes now :)

File: R/Rules-class.R (hunk 2 above)
Comment (wwojciech) [reply R/Rules-class.R:715]: acha ok.

File: R/Rules-class.R (hunk 1 above)
Comment (danielinteractive) [reply R/Rules-class.R:39]: you can try find and replace across files, tip:

Replace a string with another one in multiple files in this directory:
`perl -pi -w -e 's/OldText/NewText/g;' *.txt`

**Review by danielinteractive — APPROVED** [review]
